backend/routers/payments.py

```
@@ -22,9 +24,11 @@ async def stripe_webhook(
             payload, stripe_signature, settings.stripe_webhook_secret
         )
     except ValueError as e:
-        raise HTTPException(status_code=400, detail=str(e))
+        logger.error(f"ValueError while constructing Stripe webhook event: {e}")
+        raise HTTPException(status_code=400, detail="Invalid payload")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)